apple/native/NativeProxy.mm

```diff
@@ -175,8 +177,13 @@ static CFTimeInterval calculateTimestampWithSlowAnimations(CFTimeInterval curren
   auto maybeFlushUIUpdatesQueueFunction = [nodesManager]() { [nodesManager maybeFlushUIUpdatesQueue]; };
 
   auto requestRender = [nodesManager](std::function<void(double)> onRender, jsi::Runtime &rt) {
-    [nodesManager postOnAnimation:^(CADisplayLink *displayLink) {
+    [nodesManager postOnAnimation:^(READisplayLink *displayLink) {
+#if !TARGET_OS_OSX
       double frameTimestamp = calculateTimestampWithSlowAnimations(displayLink.targetTimestamp) * 1000;
+#else
+      // TODO: get targetTimestamp on macOS
+      double frameTimestamp = calculateTimestampWithSlowAnimations(displayLink.timestamp + displayLink.duration) * 1000;
```

Comment:
Can we split this logic so that we call `calculateTimestampWithSlowAnimations` only once outside of `#ifdef`?